node/src/Unison/Runtime/Address.hs

```diff
@@ -1,24 +1,21 @@
+{-# Language DeriveGeneric #-}
 module Unison.Runtime.Address where
 
 import Data.ByteString (ByteString)
 import Data.Bytes.Serial
 import Data.SafeCopy
 import Data.Text (Text)
 import Data.Text.Encoding (decodeUtf8, encodeUtf8)
-import System.Random
-import qualified Data.ByteString as B
+import GHC.Generics (Generic)
 import qualified Data.ByteString.Base64.URL as Base64
 import qualified Data.ByteString.Builder as Builder
 import qualified Data.ByteString.Lazy as LB
 import qualified Data.Digest.Murmur64 as Murmur
 
-class Addressor a where
-  makeAddress :: ByteString -> a
+newtype Address = Address ByteString deriving (Eq,Ord,Show,Generic)
 
-data Address = Address ByteString deriving (Eq,Ord,Show)
-
-instance Addressor Address where
-  makeAddress = fromBytes . LB.toStrict
+makeAddress :: ByteString -> Address
```

Comment:
I'd probably just call this like `murmur64Hash`, as we may want to introduce other hashing functions here. But no big deal.